### DISCUSSION TITLE
Will the cli ever support forking mode?

### DISCUSSION THREAD

**Discussion by kevindigo** [renovatebot/renovate#18722] [Request Help] [answered]
I was shocked to discover that "forking" mode is not built right into the renovate cli. It seems like a simplest and obvious use case. I am aware of the forking renovate "sister project", but it only seems to support GitHub. We would want to run this on a self-hosted GitLab instance. 

Has there been a conscious decision not to support forking mode in the main cli? Is it just a question of having someone volunteer to do the work?

**Comment by viceice** [comment]
fork mode is implemented inside renovate cli. but that is purely platform dependent and only supported by GitHub platform. nobody requested that feature this for other platforms yet. so probably nobody will implement it.

**Reply by rafsawicki** [reply]
It's been requested for Gitlab [as far as in 2017](https://github.com/renovatebot/renovate/issues/840), although the issue is now locked, so it's kinda difficult to give support for this feature.

**Reply by viceice** [reply]
unlocked and reopened

**Reply by rarkins** [reply]
@rafsawicki the issue is reopened and you are now free to contribute a PR

**Comment by rarkins** [comment, marked as answer]
The CLI already supports forking mode, but forking mode is GitHub only
